src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -579,17 +579,74 @@ def process_declarations(self, parent, nodes, arg_list):
+                container = parent.symbol_table.lookup(mod_name)
+                if not isinstance(container, ContainerSymbol):
+                    raise SymbolError(
+                        "Found a USE of module '{0}' but the symbol table "
+                        "already has a non-container entry with that name "
+                        "({1}). This is invalid Fortran.".format(
+                            mod_name, str(container)))
 
             # Create a 'deferred' symbol for each element in the ONLY clause.
             if isinstance(decl.items[4], Fortran2003.Only_List):
+                if not new_container and not container.wildcard_import \
+                   and not parent.symbol_table.imported_symbols(container):
+                    # TODO #11 Log the fact that this explicit symbol import
+                    # will replace a previous import with an empty only-list.
+                    pass
                 for name in decl.items[4].items:
-                    parent.symbol_table.add(
-                        DataSymbol(str(name).lower(),
-                                   datatype=DataType.DEFERRED,
-                                   interface=GlobalInterface(container)))
+                    # The DataSymbol adds itself to the list of symbols
+                    # imported by the Container referenced in the
+                    # GlobalInterface.
+                    sym_name = str(name).lower()
+                    if sym_name not in parent.symbol_table:
+                        parent.symbol_table.add(
+                            DataSymbol(sym_name,
+                                       datatype=DataType.DEFERRED,
+                                       interface=GlobalInterface(container)))
+                    else:
+                        # There's already a symbol with this name
+                        existing_symbol = parent.symbol_table.lookup(sym_name)
+                        if not existing_symbol.is_global:
+                            raise SymbolError(
+                                "Symbol '{0}' is imported from module '{1}' "
+                                "but is already present in the symbol table as"
+                                " either an argument or a local ({2}).".
+                                format(sym_name, mod_name,
+                                       str(existing_symbol)))
+                        # TODO #11 Log the fact that we've already got an
+                        # import of this symbol and that will take precendence.
+            elif not decl.items[3]:
+                # We have a USE statement without an ONLY clause.
+                if (not new_container) and (not container.wildcard_import) \
+                   and (not parent.symbol_table.imported_symbols(container)):
+                    # TODO #11 Log the fact that this explicit symbol import
+                    # will replace a previous import with an empty only-list.
```

Comment:
Please make it clearer that "a previous import with an empty only-list" is being replaced, rather than "a previous import" being replaced with "an empty only-list".

Reply:
Hopefully done.